docs/proposals/metrics-api.md

```diff
@@ -0,0 +1,134 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
```

Comment:
nit: Why not update the existing doc? It is slightly confusing to have multiple docs/proposals on the same topic. 
The existing proposal also contains a lot of API details that have already been discussed multiple times through various means.